Assets/Scripts/UI/MouseOverFurnitureTypeText.cs

```diff
@@ -31,19 +31,26 @@ void Start()
             return;
         }
     }
-	
+
     // Update is called once per frame
     void Update()
     {
         Tile t = mouseController.GetMouseOverTile();
 
-        string s = "NULL";
+        if (t != null)
+        {
+            string s = "NULL";
+
+            if (t.furniture != null)
+            {
+                s = t.furniture.Name;
+            }
 
-        if (t.furniture != null)
+            myText.text = "Furniture: " + s;
```

Comment:
*I am not sure whether it works on Unity. This is just a suggestion*

```
if(t != null)
{
    myText.text = "Furniture: " + (t.furniture.Name ?? "NULL");
}
```

It does exactly the same thing than what you wrote but is shorter.
https://msdn.microsoft.com/en-us/library/ms173224.aspx